dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataelement/DataElement.java

```diff
@@ -282,6 +285,32 @@ public boolean isFileType()
         return getValueType().isFile();
     }
 
+    /**
+     * Data type for use in analytics. Both text and date types are recognized
+     * as TEXT. Everything else is recognized as NUMERIC.
+     */
+    public DataType getAnalyticsDataType()
+    {
+        return (getValueType().isText() || getValueType().isDate())
```

Comment:
given that `getValueType()` is no longer a simple field access I think it is good practice to extract a local variable if the result is needed multiple times. Otherwise one uses these "getters" and in reality this triggers a cascade of computation each time. Also this helps to see that the same value is used in the return expression.

Reply:
Done.